archinstall/lib/disk/device_model.py

```diff
@@ -765,8 +779,16 @@ def device_path(self) -> Path:
 	def add_partition(self, partition: PartitionModification):
 		self.partitions.append(partition)
 
+	def get_efi_partition(self) -> Optional[PartitionModification]:
+		"""
+		Returns the first EFI partition marked for boot.
+		Perhaps we could mark it more clearly in the PartitionModification() model instead.
+		"""
+		liltered = filter(lambda x: x.fs_type == FilesystemType.Fat32 and x.is_boot() and str(x.mountpoint) == '/boot/EFI', self.partitions)
```

Comment:
Before merging, I will remove `/boot/EFI` check, and make sure that we check for the `PARTTYPE` identifier `bc13c2ff-59e6-4262-a352-b275fd6f7172` instead, and rename it to `get_xbootldr_partition` as that's really what we're checking here.
`efi` was incorrectly chosen as a placeholder name - as /boot can be an "EFI" partition as well.